diff --git a/node_modules/@atproto/api/dist/moderation/types.d.ts b/node_modules/@atproto/api/dist/moderation/types.d.ts
index 4b1ebdd..33f6fb5 100644
--- a/node_modules/@atproto/api/dist/moderation/types.d.ts
+++ b/node_modules/@atproto/api/dist/moderation/types.d.ts
@@ -1,4 +1,4 @@
-import { AppBskyActorDefs, AppBskyFeedDefs, AppBskyNotificationListNotifications, AppBskyGraphDefs, ComAtprotoLabelDefs } from '../client/index';
+import { AppBskyActorDefs, AppBskyFeedDefs, AppBskyNotificationListNotifications, AppBskyGraphDefs, ComAtprotoLabelDefs, ChatBskyActorDefs } from '../client/index';
 import { KnownLabelValue } from './const/labels';
 export declare const CUSTOM_LABEL_VALUE_RE: RegExp;
 export interface ModerationBehavior {
@@ -32,7 +32,7 @@ export interface InterpretedLabelValueDefinition extends ComAtprotoLabelDefs.Lab
     };
 }
 export type LabelDefinitionMap = Record<KnownLabelValue, InterpretedLabelValueDefinition>;
-export type ModerationSubjectProfile = AppBskyActorDefs.ProfileViewBasic | AppBskyActorDefs.ProfileView | AppBskyActorDefs.ProfileViewDetailed;
+export type ModerationSubjectProfile = AppBskyActorDefs.ProfileViewBasic | AppBskyActorDefs.ProfileView | AppBskyActorDefs.ProfileViewDetailed | ChatBskyActorDefs.ProfileViewBasic;
 export type ModerationSubjectPost = AppBskyFeedDefs.PostView;
 export type ModerationSubjectNotification = AppBskyNotificationListNotifications.Notification;
 export type ModerationSubjectFeedGenerator = AppBskyFeedDefs.GeneratorView;
diff --git a/node_modules/@atproto/api/src/moderation/types.ts b/node_modules/@atproto/api/src/moderation/types.ts
index bbf8d84..16293e2 100644
--- a/node_modules/@atproto/api/src/moderation/types.ts
+++ b/node_modules/@atproto/api/src/moderation/types.ts
@@ -4,6 +4,7 @@ import {
   AppBskyNotificationListNotifications,
   AppBskyGraphDefs,
   ComAtprotoLabelDefs,
+  ChatBskyActorDefs,
 } from '../client/index'
 import { KnownLabelValue } from './const/labels'
 
@@ -87,6 +88,7 @@ export type ModerationSubjectProfile =
   | AppBskyActorDefs.ProfileViewBasic
   | AppBskyActorDefs.ProfileView
   | AppBskyActorDefs.ProfileViewDetailed
+  | ChatBskyActorDefs.ProfileViewBasic
 
 export type ModerationSubjectPost = AppBskyFeedDefs.PostView
 
