diff --git a/node_modules/@atproto/api/dist/moderation/decision.js b/node_modules/@atproto/api/dist/moderation/decision.js
index 6dd9d4d..59df896 100644
--- a/node_modules/@atproto/api/dist/moderation/decision.js
+++ b/node_modules/@atproto/api/dist/moderation/decision.js
@@ -67,6 +67,10 @@ class ModerationDecision {
     ui(context) {
         const ui = new ui_1.ModerationUI();
         for (const cause of this.causes) {
+            // Skip !no-unauthenticated label only if user is authenticated
+            // If userDid is present in opts, user is logged in
+            if (cause?.label?.val === '!no-unauthenticated' && this.opts?.userDid) continue;
+
             if (cause.type === 'blocking' ||
                 cause.type === 'blocked-by' ||
                 cause.type === 'block-other') {
