diff --git a/dist/validators/formats.js b/dist/validators/formats.js
index fc6236017858907db20ebcfac815e1c113374527..85f8825b69602054a677697a8fe78d64cf1fb18d 100644
--- a/dist/validators/formats.js
+++ b/dist/validators/formats.js
@@ -1,5 +1,4 @@
-import isoDatestringValidator from 'iso-datestring-validator';
-const { isValidISODateString } = isoDatestringValidator;
+import { isValidISODateString } from 'iso-datestring-validator';
 import { CID } from 'multiformats/cid';
 import { validateLanguage } from '@atproto/common-web';
 import { ensureValidAtUri, ensureValidDid, ensureValidHandle, ensureValidRecordKey, isValidNsid, isValidTid, isValidUri, } from '@atproto/syntax';
