diff --git a/src/hooks/useStableCallback.ts b/src/hooks/useStableCallback.ts
index 1c788ab72351c21c5aea178d7b416741aa940e1f..d30f330ab1f082ea0477c19e2fa05b7a0b1a3e0d 100644
--- a/src/hooks/useStableCallback.ts
+++ b/src/hooks/useStableCallback.ts
@@ -6,7 +6,7 @@ type Callback = (...args: any[]) => any;
  * https://gist.github.com/JakeCoxon/c7ebf6e6496f8468226fd36b596e1985
  */
 export const useStableCallback = (callback: Callback) => {
-  const callbackRef = useRef<Callback>();
+  const callbackRef = useRef<Callback>(undefined);
   const memoCallback = useCallback(
     (...args: any) => callbackRef.current && callbackRef.current(...args),
     []
