diff --git a/node_modules/@react-navigation/native-stack/lib/commonjs/views/NativeStackView.js b/node_modules/@react-navigation/native-stack/lib/commonjs/views/NativeStackView.js
index 57c6f74..63886ef 100644
--- a/node_modules/@react-navigation/native-stack/lib/commonjs/views/NativeStackView.js
+++ b/node_modules/@react-navigation/native-stack/lib/commonjs/views/NativeStackView.js
@@ -133,10 +133,12 @@ function NativeStackView(_ref) {
 }
 const styles = _reactNative.StyleSheet.create({
   container: {
-    flex: 1
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1
   },
   contentContainer: {
-    flex: 1
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1
   },
   backImage: {
     height: 24,
diff --git a/node_modules/@react-navigation/native-stack/lib/module/views/NativeStackView.js b/node_modules/@react-navigation/native-stack/lib/module/views/NativeStackView.js
index 1a78813..74a4baf 100644
--- a/node_modules/@react-navigation/native-stack/lib/module/views/NativeStackView.js
+++ b/node_modules/@react-navigation/native-stack/lib/module/views/NativeStackView.js
@@ -125,10 +125,12 @@ export default function NativeStackView(_ref) {
 }
 const styles = StyleSheet.create({
   container: {
-    flex: 1
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1
   },
   contentContainer: {
-    flex: 1
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1
   },
   backImage: {
     height: 24,
diff --git a/node_modules/@react-navigation/native-stack/src/views/NativeStackView.tsx b/node_modules/@react-navigation/native-stack/src/views/NativeStackView.tsx
index 1646c68..bfbbee8 100644
--- a/node_modules/@react-navigation/native-stack/src/views/NativeStackView.tsx
+++ b/node_modules/@react-navigation/native-stack/src/views/NativeStackView.tsx
@@ -182,10 +182,12 @@ export default function NativeStackView({ state, descriptors }: Props) {
 
 const styles = StyleSheet.create({
   container: {
-    flex: 1,
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1,
   },
   contentContainer: {
-    flex: 1,
+    // Disabled for https://github.com/bluesky-social/social-app/pull/2126
+    // flex: 1,
   },
   backImage: {
     height: 24,
