diff --git a/node_modules/@sentry/react-native/dist/js/utils/ignorerequirecyclelogs.js b/node_modules/@sentry/react-native/dist/js/utils/ignorerequirecyclelogs.js
index 7e0b4cd..177454c 100644
--- a/node_modules/@sentry/react-native/dist/js/utils/ignorerequirecyclelogs.js
+++ b/node_modules/@sentry/react-native/dist/js/utils/ignorerequirecyclelogs.js
@@ -3,6 +3,8 @@ import { LogBox } from 'react-native';
  * This is a workaround for using fetch on RN, this is a known issue in react-native and only generates a warning.
  */
 export function ignoreRequireCycleLogs() {
-    LogBox.ignoreLogs(['Require cycle:']);
+    try {
+        LogBox.ignoreLogs(['Require cycle:']);
+    } catch (e) {}
 }
 //# sourceMappingURL=ignorerequirecyclelogs.js.map
\ No newline at end of file
