diff --git a/cli.js b/cli.js
index d825b6457105d7526c9099d03d8452f9eb87bf63..fc89a9d9a4ca8b8e3b55c9c7933084ac02e762d9 100755
--- a/cli.js
+++ b/cli.js
@@ -306,6 +306,9 @@ for (const [assetGroupName, assets] of Object.entries(groupedAssets)) {
   if (isHermes) {
     args.push('--debug-id-reference');
   }
+  if (process.env.SENTRY_DIST) {
+    args.push('--dist', process.env.SENTRY_DIST);
+  }
   args.push(...assets);
 
   const result = spawnSync(sentryCliBin, args, {
