diff --git a/node_modules/babel-preset-expo/index.js b/node_modules/babel-preset-expo/index.js
index 2099ee3..2b9e092 100644
--- a/node_modules/babel-preset-expo/index.js
+++ b/node_modules/babel-preset-expo/index.js
@@ -105,7 +105,8 @@ module.exports = function (api, options = {}) {
       ],
     ],
     plugins: [
-      getObjectRestSpreadPlugin(),
+      // - dan: This will be disabled anyway when we upgrade Expo, but let's do it now.
+      // getObjectRestSpreadPlugin(),
       ...extraPlugins,
       getAliasPlugin(),
       [require.resolve('@babel/plugin-proposal-decorators'), { legacy: true }],
