diff --git a/node_modules/babel-preset-fbjs/plugins/inline-requires.js b/node_modules/babel-preset-fbjs/plugins/inline-requires.js
index b11fc83..e18661a 100644
--- a/node_modules/babel-preset-fbjs/plugins/inline-requires.js
+++ b/node_modules/babel-preset-fbjs/plugins/inline-requires.js
@@ -256,6 +256,7 @@ function getInlineableModule(path, state) {
 
   return moduleName == null ||
     state.ignoredRequires.has(moduleName) ||
+    moduleName.startsWith('@babel/runtime/') ||
     isRequireInScope
     ? null
     : { moduleName, requireFnName: fnName };
