diff --git a/node_modules/expo-glass-effect/ios/GlassContainer.swift b/node_modules/expo-glass-effect/ios/GlassContainer.swift
index 61fb67c..6f6e9b7 100644
--- a/node_modules/expo-glass-effect/ios/GlassContainer.swift
+++ b/node_modules/expo-glass-effect/ios/GlassContainer.swift
@@ -46,11 +46,11 @@ public final class GlassContainer: ExpoView {
     }
   }
 
-  public override func mountChildComponentView(_ childComponentView: UIView, index: Int) {
+  public func mountChildComponentView(_ childComponentView: UIView, index: Int) {
     containerEffectView.contentView.insertSubview(childComponentView, at: index)
   }
 
-  public override func unmountChildComponentView(_ childComponentView: UIView, index: Int) {
+  public func unmountChildComponentView(_ childComponentView: UIView, index: Int) {
     childComponentView.removeFromSuperview()
   }
 }
diff --git a/node_modules/expo-glass-effect/ios/GlassView.swift b/node_modules/expo-glass-effect/ios/GlassView.swift
index 35cd8f3..0391b7f 100644
--- a/node_modules/expo-glass-effect/ios/GlassView.swift
+++ b/node_modules/expo-glass-effect/ios/GlassView.swift
@@ -271,11 +271,11 @@ public final class GlassView: ExpoView {
       #endif
     }
   }
-  public override func mountChildComponentView(_ childComponentView: UIView, index: Int) {
+  public func mountChildComponentView(_ childComponentView: UIView, index: Int) {
     glassEffectView.contentView.insertSubview(childComponentView, at: index)
   }
 
-  public override func unmountChildComponentView(_ childComponentView: UIView, index: Int) {
+  public func unmountChildComponentView(_ childComponentView: UIView, index: Int) {
     childComponentView.removeFromSuperview()
   }
 }
