diff --git a/node_modules/expo-updates/ios/EXUpdates/Update/NewUpdate.swift b/node_modules/expo-updates/ios/EXUpdates/Update/NewUpdate.swift
index 189a5f5..8d5b8e6 100644
--- a/node_modules/expo-updates/ios/EXUpdates/Update/NewUpdate.swift
+++ b/node_modules/expo-updates/ios/EXUpdates/Update/NewUpdate.swift
@@ -68,13 +68,20 @@ public final class NewUpdate: Update {
       processedAssets.append(asset)
     }

+    // Instead of relying on various hacks to get the correct format for the specific
+    // platform on the backend, we can just add this little patch..
+    let dateFormatter = DateFormatter()
+    dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+    dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+    let date = dateFormatter.date(from:commitTime) ?? RCTConvert.nsDate(commitTime)!
+
     return Update(
       manifest: manifest,
       config: config,
       database: database,
       updateId: uuid,
       scopeKey: config.scopeKey,
-      commitTime: RCTConvert.nsDate(commitTime),
+      commitTime: date,
       runtimeVersion: runtimeVersion,
       keep: true,
       status: UpdateStatus.StatusPending,
