diff --git a/node_modules/expo-font/ios/FontLoaderModule.swift b/node_modules/expo-font/ios/FontLoaderModule.swift
index 183480f..d1bdfb1 100644
--- a/node_modules/expo-font/ios/FontLoaderModule.swift
+++ b/node_modules/expo-font/ios/FontLoaderModule.swift
@@ -2,10 +2,23 @@ import ExpoModulesCore
 
 public final class FontLoaderModule: Module {
   // could be a Set, but to be able to pass to JS we keep it as an array
-  private var registeredFonts: [String]
+  private var _registeredFonts: [String] = []
+  private var hasQueriedNativeFonts = false
+
+  private var registeredFonts: [String] {
+    get {
+      if !hasQueriedNativeFonts {
+        hasQueriedNativeFonts = true
+        _registeredFonts = queryCustomNativeFonts()
+      }
+      return _registeredFonts
+    }
+    set {
+      _registeredFonts = newValue
+    }
+  }
 
   public required init(appContext: AppContext) {
-    self.registeredFonts = queryCustomNativeFonts()
     super.init(appContext: appContext)
   }
 
