diff --git a/node_modules/metro-transform-worker/src/index.js b/node_modules/metro-transform-worker/src/index.js
index 9f2e3d2..5222c8e 100644
--- a/node_modules/metro-transform-worker/src/index.js
+++ b/node_modules/metro-transform-worker/src/index.js
@@ -189,6 +189,10 @@ async function transformJS(file, { config, options, projectRoot }) {
   let dependencyMapName = "";
   let dependencies;
   let wrappedAst;
+  const minify =
+    options.minify &&
+    options.unstable_transformProfile !== "hermes-canary" &&
+    options.unstable_transformProfile !== "hermes-stable";
 
   // If the module to transform is a script (meaning that is not part of the
   // dependency graph and it code will just be prepended to the bundle modules),
@@ -228,19 +232,20 @@ async function transformJS(file, { config, options, projectRoot }) {
     if (config.unstable_disableModuleWrapping === true) {
       wrappedAst = ast;
     } else {
+      let moduleFactoryName;
+      if (options.dev && !minify) {
+        moduleFactoryName = file.filename;
+      }
       ({ ast: wrappedAst } = JsFileWrapping.wrapModule(
         ast,
         importDefault,
         importAll,
         dependencyMapName,
-        config.globalPrefix
+        config.globalPrefix,
+        moduleFactoryName
       ));
     }
   }
-  const minify =
-    options.minify &&
-    options.unstable_transformProfile !== "hermes-canary" &&
-    options.unstable_transformProfile !== "hermes-stable";
   const reserved = [];
   if (config.unstable_dependencyMapReservedName != null) {
     reserved.push(config.unstable_dependencyMapReservedName);
