diff --git a/node_modules/react-native-mmkv/src/createMMKV.ts b/node_modules/react-native-mmkv/src/createMMKV.ts
index 15d29a0..26ebfe9 100644
--- a/node_modules/react-native-mmkv/src/createMMKV.ts
+++ b/node_modules/react-native-mmkv/src/createMMKV.ts
@@ -48,7 +48,7 @@ export const createMMKV = (config: MMKVConfiguration): NativeMMKV => {
     // Check if we are running on-device (JSI)
     if (global.nativeCallSyncHook == null || MMKVModule.install == null) {
       throw new Error(
-        'Failed to create a new MMKV instance: React Native is not running on-device. MMKV can only be used when synchronous method invocations (JSI) are possible. If you are using a remote debugger (e.g. Chrome), switch to an on-device debugger (e.g. Flipper) instead.'
+        `Failed to create a new MMKV instance: React Native is not running on-device. MMKV can only be used when synchronous method invocations (JSI) are possible. If you are using a remote debugger (e.g. Chrome), switch to an on-device debugger (e.g. Flipper) instead. global.nativeCallSyncHook == null: ${global.nativeCallSyncHook == null}; MMKVModule.install == null: ${MMKVModule.install == null}`
       );
     }
 
@@ -67,4 +67,4 @@ export const createMMKV = (config: MMKVConfiguration): NativeMMKV => {
   }
 
   return global.mmkvCreateNewInstance(config);
-};
+};z
