diff --git a/ios/RNDatePicker.h b/ios/RNDatePicker.h
index 480746eb7acfbe86f67547d9e1de7a5be4d5faf2..13d30cb547195993dfcb4005cc0d248de9ac391a 100644
--- a/ios/RNDatePicker.h
+++ b/ios/RNDatePicker.h
@@ -15,6 +15,7 @@ NS_ASSUME_NONNULL_END
 #else
 #import "DatePicker.h"
 #import <UIKit/UIKit.h>
+#include <string>
 
 @interface RNDatePicker : DatePicker
 
@@ -22,4 +23,3 @@ NS_ASSUME_NONNULL_END
 @end
 
 #endif
-
