diff --git a/node_modules/react-native-mmkv/lib/commonjs/createMMKV.js b/node_modules/react-native-mmkv/lib/commonjs/createMMKV.js
index b9b9d69..e85a178 100644
--- a/node_modules/react-native-mmkv/lib/commonjs/createMMKV.js
+++ b/node_modules/react-native-mmkv/lib/commonjs/createMMKV.js
@@ -39,7 +39,7 @@ const createMMKV = config => {
 
     // Check if we are running on-device (JSI)
     if (global.nativeCallSyncHook == null || MMKVModule.install == null) {
-      throw new Error('Failed to create a new MMKV instance: React Native is not running on-device. MMKV can only be used when synchronous method invocations (JSI) are possible. If you are using a remote debugger (e.g. Chrome), switch to an on-device debugger (e.g. Flipper) instead.');
+      throw new Error(`Failed to create a new MMKV instance: React Native is not running on-device etc etc. typeof global.nativeCallSyncHook: ${typeof global.nativeCallSyncHook}, typeof MMKVModule.install: ${typeof MMKVModule.install} (CJS)`);
     }
 
     // Call the synchronous blocking install() function
diff --git a/node_modules/react-native-mmkv/lib/module/createMMKV.js b/node_modules/react-native-mmkv/lib/module/createMMKV.js
index 644840d..8ebb03e 100644
--- a/node_modules/react-native-mmkv/lib/module/createMMKV.js
+++ b/node_modules/react-native-mmkv/lib/module/createMMKV.js
@@ -33,7 +33,7 @@ export const createMMKV = config => {
 
     // Check if we are running on-device (JSI)
     if (global.nativeCallSyncHook == null || MMKVModule.install == null) {
-      throw new Error('Failed to create a new MMKV instance: React Native is not running on-device. MMKV can only be used when synchronous method invocations (JSI) are possible. If you are using a remote debugger (e.g. Chrome), switch to an on-device debugger (e.g. Flipper) instead.');
+      throw new Error(`Failed to create a new MMKV instance: React Native is not running on-device etc etc. typeof global.nativeCallSyncHook: ${typeof global.nativeCallSyncHook}, typeof MMKVModule.install: ${typeof MMKVModule.install} (MJS)`);
     }
 
     // Call the synchronous blocking install() function
diff --git a/node_modules/react-native-mmkv/src/createMMKV.ts b/node_modules/react-native-mmkv/src/createMMKV.ts
index 15d29a0..150e53e 100644
--- a/node_modules/react-native-mmkv/src/createMMKV.ts
+++ b/node_modules/react-native-mmkv/src/createMMKV.ts
@@ -48,7 +48,7 @@ export const createMMKV = (config: MMKVConfiguration): NativeMMKV => {
     // Check if we are running on-device (JSI)
     if (global.nativeCallSyncHook == null || MMKVModule.install == null) {
       throw new Error(
-        'Failed to create a new MMKV instance: React Native is not running on-device. MMKV can only be used when synchronous method invocations (JSI) are possible. If you are using a remote debugger (e.g. Chrome), switch to an on-device debugger (e.g. Flipper) instead.'
+        `Failed to create a new MMKV instance: React Native is not running on-device etc etc. typeof global.nativeCallSyncHook: ${typeof global.nativeCallSyncHook}, typeof MMKVModule.install: ${typeof MMKVModule.install}`
       );
     }
 
