diff --git a/node_modules/sonner-native/lib/commonjs/toast.js b/node_modules/sonner-native/lib/commonjs/toast.js
index 121816a..ed932f5 100644
--- a/node_modules/sonner-native/lib/commonjs/toast.js
+++ b/node_modules/sonner-native/lib/commonjs/toast.js
@@ -265,6 +265,7 @@ const Toast = exports.Toast = /*#__PURE__*/React.forwardRef(({
       children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativeReanimated.default.View, {
         entering: entering,
         exiting: exiting,
+        collapsible: false,
         children: jsx
       })
     });
@@ -275,6 +276,7 @@ const Toast = exports.Toast = /*#__PURE__*/React.forwardRef(({
       style: [unstyled ? undefined : elevationStyle, defaultStyles.toast, toastStyleCtx, styles?.toast, style, wiggleAnimationStyle],
       entering: entering,
       exiting: exiting,
+      collapsible: false,
       children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactNative.View, {
         style: [defaultStyles.toastContent, toastContentStyleCtx, styles?.toastContent],
         children: [promiseOptions || variant === 'loading' ? 'loading' in icons ? icons.loading : /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.ActivityIndicator, {}) : icon ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
